initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/container/dockercompose/DockerComposeFile.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.spring.container.dockercompose;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * A Docker Compose file.
+ *
+ * @author Moritz Halbritter
+ */
+public class DockerComposeFile {
```

Comment:
The build infrastructure has this notion of container where you can create or get an existing item with id and further customize it. I don't know if that's relevant here.

The service is immutable (as it should) but if we had a builder pattern, this would let contributions to add something to the environment, for instance.

I wonder about `File` as well. `DockerCompose` feels unqualified but I wonder if we couldn't find another name for this.

Reply:
I've keyed the `DockerComposeFile` by service name. Adding a service with the same name overrides the old one, like the `DependencyContainer` does.

Reply:
It’s not a container of docker compose but a container of services. It doesn’t have to have the suffit containerI guess.